modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -634,6 +642,66 @@ public boolean isSnapshotCreating() {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public IgniteFuture<Void> cancelSnapshot(String name) {
+        A.notNullOrEmpty(name, "Snapshot name must be not empty or null");
+
+        IgniteInternalFuture<Void> fut0 = cctx.kernalContext().closure()
+            .broadcast(new CancelSnapshotClosure(),
```

Comment:
Suppose that next processes in the cluster will happen simultaneously:
1. The `END_SNAPSHOT` DP will be at the finish stage and only part of nodes processed their 'this::processLocalSnapshotEndStageResult'
2. Part of nodes executes broadcasted closure `cancelLocalSnapshotTask`

They are processed by different threads. Can we get a situation when part of nodes finish `clusterSnpFut` successfully and other ones with error? It's ok?

Reply:
Nikita,

1. It's true that some operations may execute concurrently on snapshot cancellation operation. 
Snapshot operation consists of four stages:
- start-stage 
- start-stage result
- end-stage
- end-stage result

If cancel operation occurred when on start-stage than snapshot operation will be canceled. On other stages, it will have no effect since all the cache data already collected.

2. `clusterSnpFut` is set on coordinator only, so it has no effect in case described by you. If some of the local snapshot tasks finish with an error an the other with `CancelledException` than all results will be finished with `CancelledException` (all intermediate results will be deleted).

Reply:
Ok, got it.